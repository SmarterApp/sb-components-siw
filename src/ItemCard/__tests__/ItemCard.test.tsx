import * as React from "react";
import { shallow } from "enzyme";
import { ItemCard, ItemCardModel, GradeLevels } from "@src/index";
import { itemCardProps } from "@mocks/index";

describe("ItemCard", () => {
  const wrapper = shallow(<ItemCard {...itemCardProps} />);

  it("renders correctly", () => {
    expect(wrapper).toMatchSnapshot();
  });
});
