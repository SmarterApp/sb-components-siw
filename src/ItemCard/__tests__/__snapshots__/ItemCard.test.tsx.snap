// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ItemCard renders correctly 1`] = `
<React.Fragment>
  <div
    className="card card-block ela"
    onClick={[Function]}
    onKeyUp={[Function]}
    role="link"
    tabIndex={0}
  >
    <div
      className="card-contents"
    >
      <div
        className="card-header"
      >
        <h4
          className="card-title"
        >
          ELA
        </h4>
        <div
          className="card-icon-container"
        >
          <span
            className="card-grade-tag card-icon"
          >
            G5
          </span>
        </div>
      </div>
      <p
        className="card-text grade"
      >
        <span
          className="card-text-label"
        >
          Grade:
        </span>
        <span
          className="card-text-value"
        >
           
          Grade 4
        </span>
      </p>
      <React.Fragment>
        <p
          className="card-text test-name"
        >
          <span
            className="card-text-label"
          >
            Test name:
          </span>
          <span
            className="card-text-value"
          >
             
          </span>
        </p>
        <p
          className="card-text item-position-in-test"
        >
          <span
            className="card-text-label"
          >
            Item position in test:
          </span>
          <span
            className="card-text-value"
          >
             
          </span>
        </p>
      </React.Fragment>
      <p
        className="card-text stimulusid"
      >
        <span
          className="card-text-label"
        >
          Stimulus ID:
        </span>
        <span
          className="card-text-value"
        >
           
        </span>
      </p>
      <p
        className="card-text claim"
      >
        <span
          className="card-text-label"
        >
          Claim:
        </span>
        <span
          className="card-text-value"
        >
           
          Claim
        </span>
      </p>
      <p
        className="card-text target"
      >
        <span
          className="card-text-label"
        >
          Target:
        </span>
        <span
          className="card-text-value"
        >
          <ToolTip
            className="box"
            displayIcon={true}
            displayText="dfas"
            helpText={
              <span>
                target description right here
              </span>
            }
          />
        </span>
      </p>
      <p
        className="card-text target"
      >
        <span
          className="card-text-label"
        >
          Standard:
        </span>
        <span
          className="card-text-value"
        >
          <ToolTip
            className="box"
            displayIcon={true}
            displayText="AB-123"
            helpText={
              <span>
                this is the common core description
              </span>
            }
          />
        </span>
      </p>
      <p
        className="card-text interaction-type"
      >
        <span
          className="card-text-label"
        >
          Item Type:
        </span>
        <span
          className="card-text-value"
        >
           Item Type
        </span>
      </p>
      <p
        className="card-text item-id"
      >
        <span
          className="card-text-label"
        >
          Item Id:
        </span>
        <span
          className="card-text-value"
        >
           
          3434
        </span>
      </p>
      <ToolTip
        className="tooltip-item-card-button"
        helpText={
          <span>
            Add item to print queue
          </span>
        }
        includeTabindex={true}
        onKeyPress={[Function]}
      >
        <button
          aria-hidden={true}
          className="btn btn-add-remove-print-selection ela  btn-unselected "
          onClick={[Function]}
          tabIndex={-1}
          type="button"
        >
          <i
            className="fa  fa-plus-square"
          />
            
           Select to Print
        </button>
      </ToolTip>
    </div>
  </div>
</React.Fragment>
`;
